tests/tokenWrapper.test.ts

```diff
@@ -0,0 +1,800 @@
+        let newIsNativeAllowed = false;
+        let newWrappingLimit = 1000;
+        let newFeePercentage = 2;
+        let newFeeRecipient = FerdieSigner.address;
+
+        // update config with new states
+        let updateConfigFunction = await tokenWrapperContract.tx.updateConfig(newGovernor, newIsNativeAllowed, newWrappingLimit,
+            newFeePercentage, newFeeRecipient)
+        expect(updateConfigFunction).to.be.ok;
+
+        // validate that new governor is not same as old governor
+        let newGovernorFromStorage = await tokenWrapperContract.query.governor();
+        expect(newGovernorFromStorage.output).to.equal(newGovernor);
+        expect(newGovernorFromStorage.output).to.not.equal(contractGovernor);
+
+        // validate that isNativeAllowed has changed
+        let  newIsNativeAllowedFromStorage = await tokenWrapperContract.query.isNativeAllowed();
+        expect(newIsNativeAllowedFromStorage.output).to.be.equal(newIsNativeAllowed);
+        expect(newIsNativeAllowedFromStorage.output).to.not.equal(isNativeAllowed);
+
+        // validate that isWrappingLimit has changed
+        let newWrappingLimitFromStorage = await tokenWrapperContract.query.wrappingLimit();
+        expect(Number(newWrappingLimitFromStorage.output)).to.equal(newWrappingLimit);
+        expect(Number(newWrappingLimitFromStorage.output)).to.not.equal(wrappingLimit);
+
+        // validate that feePercentage has changed
+        let newFeePercentageFromStorage = await tokenWrapperContract.query.feePercentage();
+        expect(Number(newFeePercentageFromStorage.output)).to.equal(newFeePercentage);
+        expect(Number(newFeePercentageFromStorage.output)).to.not.equal(feePercentage);
+
+        // validate that feeRecipient has changed
+        let newFeeRecipientFromStorage = await tokenWrapperContract.query.feeRecipient();
+        expect(newFeeRecipientFromStorage.output).to.equal(newFeeRecipient);
+        expect(newFeeRecipientFromStorage.output).to.not.equal(feeRecipient);
+
+    });
+
+    it('Test native wrapping functionality', async () => {
+        let initialSenderWrappedBalance = await tokenWrapperContract.query.psp22Balance(sender.address);
+        let initialContractBalance = await tokenWrapperContract.query.nativeContractBalance();
+
+        expect(Number(initialSenderWrappedBalance.output)).to.not.equal(0);
+        expect(Number(initialContractBalance.output)).to.equal(0);
+
+        let wrapFunction = await tokenWrapperContract.tx.wrap( null, 10, { value: 1500 });
+
+        expect(wrapFunction).to.be.ok;
+
+        // to validate that psp22 token has been minted for sender
+        let senderWrappedBalanceAfter = await tokenWrapperContract.query.psp22Balance(sender.address);
+        // to validate that native funds was transferred to the contract
+        let contractBalanceAfter = await tokenWrapperContract.query.nativeContractBalance();
+
+        expect(Number(senderWrappedBalanceAfter.output)).to.not.equal(0);
```

Comment:
After we wrap more tokens, we need to verify that the balance increased by that amount. Why is the check only that it is not zero? If it was not zero before wrapping why would this test anything?

Reply:
There is actually checks to verify that amount increased, see
https://github.com/webb-tools/protocol-ink/blob/dami/governed-token-wrapper-test/tests/tokenWrapper.test.ts#L263-L264